torch/gpu.py

```diff
@@ -0,0 +1,82 @@
+from torch.device_interface import DeviceInterface, get_interface_for_device
+
+
+# TODO: OOB. torch.cuda.current_stream()  -->  torch.gpu.current_stream(device_type='cuda')
+# TODO: OOB. torch.cuda.Event()  -->  torch.gpu.Event(device_type='cuda')
+# TODO: OOB. torch.cuda.synchronize()  -->  torch.gpu.synchronize(device_type='cuda')
+# TODO: name change?
+# TODO: location change?
+class gpu:
```

Comment:
Not sure about the necessity of adding this new frontend API here. I guess we basically want a device runtime abstraction that can be extended to support new devices and with the device interfaces aligning with CUDA? This can be achieved with `DeviceInterface` and the registration infrastructure without this new "gpu" API?